src/vscode-cosmosdb.api.d.ts

```diff
@@ -0,0 +1,30 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export interface CosmosDBDatabase {
+    accountName: string
+    connectionString: string
+    treeItemId: string
+    databaseName: string | undefined
+}
+
+export interface VSCodeCosmosDB {
+    /**
+     * Finds the database in CosmosDB and returns CosmosDBDatabase object or undefined if can't find
+     * @param detectionData The database connection string
+     */
+    getDatabase(detectionData: { connectionString: string }): Promise<CosmosDBDatabase | undefined>;
+
+    /**
+     *  Traverses the CosmosDB tree with a quick pick at each level. Goes until find item with database-level context value. Returns the CosmosDBDatabase object based on picked db.
+     */
+    pickDatabase(): Promise<CosmosDBDatabase | undefined>;
```

Comment:
Our other UI methods throw UserCancelled instead of returning undefined.  Is there a reason this one doesn't do the same?

Reply:
I asked him to do this. I thought it didn't make sense to "expose" UserCancelledError through an "official" api since other teams might use this eventually. I figured it was better to follow the VS Code pattern which just returns undefined.

Reply:
Okay. Let's doc that behavior, then.